analyzers/src/SonarAnalyzer.Common/Rules/ParametersCorrectOrderBase.cs

```diff
@@ -117,11 +129,23 @@ public abstract class ParametersCorrectOrderBase<TArgumentSyntax> : SonarDiagnos
                 argumentIdentifiers.Any(ia =>
                     ia.IdentifierName == parameter.Name &&
                     GetArgumentTypeSymbolInfo(ia.ArgumentSyntax, semanticModel).ConvertedType.DerivesOrImplements(parameter.Type));
+
+            bool IdentifierWithSameNameAndTypeExistsLater(ArgumentIdentifier argumentIdentifier, int index) =>
+                argumentIdentifiers.Skip(index + 1)
+                                   .Any(ia => string.Equals(ia.IdentifierName, argumentIdentifier.IdentifierName, StringComparison.OrdinalIgnoreCase)
+                                                              && ArgumentTypesAreSame(ia.ArgumentSyntax, argumentIdentifier.ArgumentSyntax));
```

Comment:
What was the motivation for having this method? AFAICT it prevents the issue from being raised twice when arguments are swapped. For sake of simplicity, I'd say it's fine to raise twice and we can remove this check.

Reply:
Without this method there is different result for same test. Example is passing one var to both params, see TestCases\ParametersCorrectOrder.cs line 97 and 101. (pass "a" and "b" to both params of `m(int a, int b)`)
I think, that if argument was passed at correct place once, we shouldn't rise

Reply:
Ok, that makes sense. And at the same time, if `b` was passed to both, we would have an equal scenario and we would raise and we shouldn't, right?
So we need to know if the argument was passed on the correct location at least once (no matter if before or after). This sounds complicated to do in a clean way to me. To keep this PR focused, I would be fine with creating a new issue for that case, fix it in another PR and here just document those two cases `m(a, a)` and `m(b, b)` as `' Noncompliant FP` with a link to that new issue.